pymodels/TS_V04_01/lattice.py

```diff
@@ -0,0 +1,262 @@
+
+import math as _math
+import pyaccel as _pyaccel
+import numpy as _np
+import mathphys as _mp
+from . import segmented_models as _segmented_models
+
+
+class LatticeError(Exception):
+    pass
+
+energy = 3e9  #[eV]
+default_optics_mode = 'M1'
+
+def create_lattice(optics_mode = default_optics_mode):
+
+    strengths, twiss_at_start = get_optics_mode(optics_mode)
+
+    # -- shortcut symbols --
+    marker       = _pyaccel.elements.marker
+    drift        = _pyaccel.elements.drift
+    quadrupole   = _pyaccel.elements.quadrupole
+    rbend_sirius = _pyaccel.elements.rbend
+    sextupole    = _pyaccel.elements.sextupole
+    hcorrector   = _pyaccel.elements.hcorrector
+    vcorrector   = _pyaccel.elements.vcorrector
+
+    corr_length = 0.07
+
+    # --- drift spaces ---
+    ldif = 0.1442
+    l015 = drift('l015', 0.15)
+    l020 = drift('l020', 0.20)
+    l025 = drift('l025', 0.25)
+    l060 = drift('l060', 0.60)
+    l080 = drift('l080', 0.80)
+    l090 = drift('l090', 0.90)
+    l160 = drift('l160', 1.60)
+    l280 = drift('l280', 2.80)
+    l400 = drift('l400', 4.00)
+    la2p = drift('la2p', 0.08323)
+    la3p = drift('la3p', 0.232-ldif)
+    lb1p = drift('lb1p', 0.220-ldif)
+    lb2p = drift('lb2p', 0.83251)
+    lb3p = drift('lb3p', 0.30049)
+    lb4p = drift('lb4p', 0.19897-ldif)
+    lc1p = drift('lc1p', 0.18704-ldif)
+    lc2p = drift('lc2p', 0.07304)
+    lc3p = drift('lc3p', 0.19934)
+    lc4p = drift('lc4p', 0.72666-ldif)
+    ld1p = drift('ld1p', 0.25700-ldif)
+    ld2p = drift('ld2p', 0.05389)
+    ld3p = drift('ld3p', 0.154)
+    ld4p = drift('ld4p', 0.192)
+    ld5p = drift('ld5p', 0.456)
+    ld6p = drift('ld6p', 0.258)
+    ld7p = drift('ld7p', 0.175)
+
+    # --- markers ---
+    inicio = marker('start')
+    fim    = marker('end')
+
+    # --- beam screens ---
+    scrn   = marker('Scrn')
+
+    # --- beam current monitors ---
+    ict    = marker('ICT')
+    fct    = marker('FCT')
+
+    # --- beam position monitors ---
+    bpm    = marker('BPM')
+
+    # --- correctors ---
+    ch     = hcorrector('CH', 0.0)
+    cv     = vcorrector('CV', 0.0)
```

Comment:
I think it's better to model the correctors with a thick lens element. This is the way we do it in the other accelerators.
The fact that some of them are inside quadrupoles is not a problem as long as you define their indices in the family_data module

Reply:
Done